spec/lib/accession/sample_spec.rb

```diff
@@ -145,4 +160,70 @@
     expect(sample.update_accession_number('ENA1234')).to be_truthy
     expect(sample.ebi_accession_number).to eq('ENA1234')
   end
+
+  describe '#to_xml' do
+    context 'with country of origin' do
+      it 'includes country of origin' do
+        sample = described_class.new(tag_list, create(:sample_for_accessioning_with_open_study))
+        expect(sample.to_xml).to include(COUNTRY_TAG)
+        expect(find_value_at_tag(sample.to_xml, COUNTRY_TAG)).to eq('Australia')
+      end
+
+      it 'displays not collected when country is empty' do
+        smpl = create(:sample_for_accessioning_with_open_study)
+        smpl.sample_metadata.update(country_of_origin: nil)
+        sample = described_class.new(tag_list, smpl)
+        expect(sample.to_xml).to include(COUNTRY_TAG)
+        expect(find_value_at_tag(sample.to_xml, COUNTRY_TAG)).to eq('not collected')
+      end
+
+      it 'displays not provided when country value is not provided' do
+        smpl = create(:sample_for_accessioning_with_open_study)
+        smpl.sample_metadata.update(country_of_origin: 'not provided')
+        sample = described_class.new(tag_list, smpl)
+        expect(sample.to_xml).to include(COUNTRY_TAG)
+        expect(find_value_at_tag(sample.to_xml, COUNTRY_TAG)).to eq('not provided')
+      end
+
+      it 'displays not collected when country value is wrong' do
+        smpl = create(:sample_for_accessioning_with_open_study)
+        smpl.sample_metadata.update(country_of_origin: 'Freedonia')
+        sample = described_class.new(tag_list, smpl)
+        expect(sample.to_xml).to include(COUNTRY_TAG)
+        expect(find_value_at_tag(sample.to_xml, COUNTRY_TAG)).to eq('not collected')
+      end
+    end
+
+    context 'with collection date' do
+      it 'includes collection date' do
+        sample = described_class.new(tag_list, create(:sample_for_accessioning_with_open_study))
+        expect(sample.to_xml).to include(COLLECTION_DATE_TAG)
+        expect(find_value_at_tag(sample.to_xml, COLLECTION_DATE_TAG)).to eq('2000-01-01T00:00')
```

Comment:
Do the date-times need to have ISO separator 'T' ?
Can date-times have missing minutes part?
Can date-times have date and a trailing T only, for example '2000-01-01T' ?